### ISSUE TITLE
Bad error message when Zellij cannot create its temporary directory

### ISSUE THREAD

**Issue by vi** [zellij-org/zellij#2453]
```
$ zellij --version
zellij 0.36.0

$ rm -Rf /tmp/zellij-1000/
$ touch /tmp/zellij-1000
$ chmod 000 /tmp/zellij-1000

$ zellij --version
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 20, kind: NotADirectory, message: "Not a directory" }', /home/vi/.cargo/registry/src/-3d9d141e372ea94e/zellij-utils-0.36.0/src/logging.rs:28:45
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

1. `zellij --version` and `zellij --help` should not mutate filesystem at all and should not try to open any config files (unless really needed e.g. for showing plugins' help).
2. Failure to access or create the Zellij home directory should be explicitly handled and a user-friendly message should be displayed, mentioning config file location and config file parameter to override the Zellij home directory location, preferably also with an environment variable e.g. `ZELLIJ_HOME` to do the same.

**Comment by qballer** [comment]
I think ` /tmp/zellij-1000` is a folder.

**Comment by vi** [comment]
Obviously, `/tmp/zellij-1000` is supposed to be a directory. The commands above are just an easy way to make `/tmp/zellij-1000` unavailable to Zellij.

Other cases where `$TMPDIR` may fail to work includes:

* `/tmp/` is mounted read-only
* `/tmp/zellij-1000` is owned by another uid (not 1000)
* Quota limits preventing using `/tmp`
* Non-standard distro without a working `/tmp`
* `/tmp/` on NFS and network is down

Zellij should not panic in those cases, but show some user-friendly message.

Temporary directory for Zellij can already be overridden with `TMPDIR` environment variable, but Zellij-specific one (pointing to thing like `/tmp/zellij-1000` directly, not to `/tmp`) would also help.
